### ISSUE TITLE
Incorrect types for 'defer'

### ISSUE THREAD

**Issue by JClackett** [vercel/remix#42]
Hey, for remix versions 1.17.1, I noticed that the type returned from the useLoaderData after using defer instead of json gives some weird typings.

```ts
// spot =  {id: string, etc...), rating = number
return defer(
  { spot, rating },
)


const promise = useLoaderData<typeof loader>()
  // const promise: {
  //   data: {
  //       spot: {
  //           readonly [Symbol.toStringTag]: string;
  //       };
  //       rating: {
  //           readonly [Symbol.toStringTag]: string;
  //       };
  //   };
  //   readonly [typedDeferredDataBrand]: "TypedDeferredData";
  //   init?: {
  //       headers?: [...][] | ... 2 more ... | undefined;
  //       status?: number | undefined;
  //       statusText?: string | undefined;
  //   } | undefined;
  // }


<React.Suspense fallback={<SpotFallback />}>
        <Await resolve={promise.spot}>
// Property 'spot' does not exist on type '{ data: { spot: { readonly [Symbol.toStringTag]: string; }; rating: { readonly [Symbol.toStringTag]: string; }; }; readonly [typedDeferredDataBrand]: "TypedDeferredData"; init?: { headers?: [...][] | ... 2 more ... | undefined; status?: number | undefined; statusText?: string | undefined; } | undefined; }'.ts(2339)
```

**Comment by JClackett** [comment]
and if I use the defer from @remix-run/node its fine

**Comment by titong0** [comment]
Facing the same issue. was there anything you could do to solve it? I'm thinking about patching the type but it doesn't feel right

**Comment by BCarroll524** [comment]
i'm experiencing the same issue. Would be great if we can get the typings fixed otherwise just going to keep using `defer` from the `@vercel/remix` package but have `@ts-expect-error` for now

**Comment by titong0** [comment]
Sorry for not following this up, I fixed it by updating remix, if you installed @vercel/remix recently then it it is likely on a newer version, incompatible with your existing remix project.

**Comment by MonsterDeveloper** [comment]
Same here. @titong0 did you manage to solve it?

![CleanShot 2023-12-01 at 12 43 32@2x](https://github.com/vercel/remix/assets/16190582/59bd4d9d-c066-462c-b168-98b9f451bb10)

[CodeSandbox](https://codesandbox.io/p/devbox/vibrant-bessie-qj7dz8?layout=%257B%2522sidebarPanel%2522%253A%2522EXPLORER%2522%252C%2522rootPanelGroup%2522%253A%257B%2522direction%2522%253A%2522horizontal%2522%252C%2522contentType%2522%253A%2522UNKNOWN%2522%252C%2522type%2522%253A%2522PANEL_GROUP%2522%252C%2522id%2522%253A%2522ROOT_LAYOUT%2522%252C%2522panels%2522%253A%255B%257B%2522type%2522%253A%2522PANEL_GROUP%2522%252C%2522contentType%2522%253A%2522UNKNOWN%2522%252C%2522direction%2522%253A%2522vertical%2522%252C%2522id%2522%253A%2522clpmjw1bk00073b6eyynrlce2%2522%252C%2522sizes%2522%253A%255B70%252C30%255D%252C%2522panels%2522%253A%255B%257B%2522type%2522%253A%2522PANEL_GROUP%2522%252C%2522contentType%2522%253A%2522EDITOR%2522%252C%2522direction%2522%253A%2522horizontal%2522%252C%2522id%2522%253A%2522EDITOR%2522%252C%2522panels%2522%253A%255B%257B%2522type%2522%253A%2522PANEL%2522%252C%2522contentType%2522%253A%2522EDITOR%2522%252C%2522id%2522%253A%2522clpmjw1bk00023b6evnuyi7cx%2522%257D%255D%257D%252C%257B%2522type%2522%253A%2522PANEL_GROUP%2522%252C%2522contentType%2522%253A%2522SHELLS%2522%252C%2522direction%2522%253A%2522horizontal%2522%252C%2522id%2522%253A%2522SHELLS%2522%252C%2522panels%2522%253A%255B%257B%2522type%2522%253A%2522PANEL%2522%252C%2522contentType%2522%253A%2522SHELLS%2522%252C%2522id%2522%253A%2522clpmjw1bk00043b6e871rocnh%2522%257D%255D%252C%2522sizes%2522%253A%255B100%255D%257D%255D%257D%252C%257B%2522type%2522%253A%2522PANEL_GROUP%2522%252C%2522contentType%2522%253A%2522DEVTOOLS%2522%252C%2522direction%2522%253A%2522vertical%2522%252C%2522id%2522%253A%2522DEVTOOLS%2522%252C%2522panels%2522%253A%255B%257B%2522type%2522%253A%2522PANEL%2522%252C%2522contentType%2522%253A%2522DEVTOOLS%2522%252C%2522id%2522%253A%2522clpmjw1bk00063b6eg2g9ancy%2522%257D%255D%252C%2522sizes%2522%253A%255B100%255D%257D%255D%252C%2522sizes%2522%253A%255B50%252C50%255D%257D%252C%2522tabbedPanels%2522%253A%257B%2522clpmjw1bk00023b6evnuyi7cx%2522%253A%257B%2522id%2522%253A%2522clpmjw1bk00023b6evnuyi7cx%2522%252C%2522tabs%2522%253A%255B%255D%257D%252C%2522clpmjw1bk00063b6eg2g9ancy%2522%253A%257B%2522tabs%2522%253A%255B%257B%2522id%2522%253A%2522clpmjw1bk00053b6e4tjkeb09%2522%252C%2522mode%2522%253A%2522permanent%2522%252C%2522type%2522%253A%2522TASK_PORT%2522%252C%2522taskId%2522%253A%2522dev%2522%252C%2522port%2522%253A3000%252C%2522path%2522%253A%2522%252F%2522%257D%255D%252C%2522id%2522%253A%2522clpmjw1bk00063b6eg2g9ancy%2522%252C%2522activeTabId%2522%253A%2522clpmjw1bk00053b6e4tjkeb09%2522%257D%252C%2522clpmjw1bk00043b6e871rocnh%2522%253A%257B%2522id%2522%253A%2522clpmjw1bk00043b6e871rocnh%2522%252C%2522activeTabId%2522%253A%2522clpmjwg5s008p3b6emuwyi43t%2522%252C%2522tabs%2522%253A%255B%257B%2522id%2522%253A%2522clpmjw1bk00033b6euehiuvr1%2522%252C%2522mode%2522%253A%2522permanent%2522%252C%2522type%2522%253A%2522TASK_LOG%2522%252C%2522taskId%2522%253A%2522dev%2522%257D%252C%257B%2522id%2522%253A%2522clpmjwg5s008p3b6emuwyi43t%2522%252C%2522mode%2522%253A%2522permanent%2522%252C%2522type%2522%253A%2522TERMINAL%2522%252C%2522shellId%2522%253A%2522clpmjwgdd001dedgefpa3ebno%2522%257D%255D%257D%257D%252C%2522showDevtools%2522%253Atrue%252C%2522showShells%2522%253Atrue%252C%2522showSidebar%2522%253Atrue%252C%2522sidebarPanelSize%2522%253A15%257D)

**Comment by JClackett** [comment]
Making sure that the vercel/remix package and the regular remix packages all have the same version is key, if they are all 2.2.0 it works for me

**Comment by JClackett** [comment]
@MonsterDeveloper Yeah my guess is that they may be working/waiting (not sure who is actually building it) on the new Vite plugin for remix as it kind of makes this package obsolete, but it would be nice to have the latest remix version until then :D

**Comment by muco-rolle** [comment]
Hello, @JClackett it seems not to be working for me, even though I have added all the packages the same as @vercel/remix (2.2.0).

The attached screenshots show that when using the vercel package and the node package, you can see that with the node package, the types are correct.

![Screenshot from 2023-12-06 00-44-04](https://github.com/vercel/remix/assets/43722815/6a0352cb-3694-467e-bdfe-49484409316a)
![Screenshot from 2023-12-06 00-44-27](https://github.com/vercel/remix/assets/43722815/9d6d1d57-d9aa-4c37-99ef-5b65c736c635)

I would appreciate it if someone could give me a hand. Here's a [reproducible StackBlitz sandbox](https://stackblitz.com/edit/remix-run-remix-9tqxff?file=app%2Froutes%2F_index.tsx).

**Comment by muco-rolle** [comment]
Update: This has been fixed with the release of `@vercel/remix 2.3.1`
